tests/network/port_forward.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = SIGDescribe("[Serial] Port-forward", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("VMI With masquerade binding", func() {
+		const localPort = 1500
+		var (
+			portForwardCmd *exec.Cmd
+			vmi            *v1.VirtualMachineInstance
+			VMIport        int
+			ports          []v1.Port
+		)
+
+		JustBeforeEach(func() {
+			vmi = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, ports)
+			tests.StartHTTPServer(vmi, VMIport)
+
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+			Expect(vmiPod).ToNot(BeNil())
+			portForwardCmd, err = portForwardCommand(vmiPod, localPort, VMIport)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(portForwardCmd.Start()).To(Succeed())
+		})
+
+		When("performing port-forward from a local port to a VMI's pod declared port", func() {
+			const declaredPort = 1501
+			BeforeEach(func() {
+				ports = append(ports, v1.Port{Port: declaredPort})
+				VMIport = declaredPort
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
+					Expect(killPortForwardCommand(portForwardCmd)).To(Succeed())
+				}()
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", declaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 40*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+
+		When("performing port-forward from a local port to a VMI's pod non-declared port", func() {
+			const nonDeclaredPort = 1502
+			BeforeEach(func() {
+				ports = []v1.Port{}
+				VMIport = nonDeclaredPort
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
```

Comment:
Why `defer` and not `AfterEach`? Since the command was created in the `BeforeEach` I believe it makes sense to kill it in the `AfterEach`.

Reply:
@rhrazdil @alonSadan I think there was a misunderstanding. The original [comment](https://github.com/kubevirt/kubevirt/pull/5200#discussion_r617235464) explained why we are using `tests.BeforeTestCleanup` and not cleaning all the entities on the `AfterEach` of the tests. The main reason is that the reports are collected after the teardown.
This is the standard. And if we have a new type of entity that is not cleaned by `tests.BeforeTestCleanup` we prefer to add it to `tests.BeforeTestCleanup` instead of doing a specific cleaning in the `AfterEach`. However, here a process is cleaned, not an entity (am I right? is there any entity created while this process is running?). I don't think we collect information about the running processes and even if we do, did you verify using `defer` will help the situation and the report is collected after it (I believe that now the cleanup we be executed even vefore the `AfterEach`)?

Reply:
You are right. There is no cleanup in the defer so i is redundant I think.
Thanks.
Moving back to afterEach.

Reply:
Ah I see, thanks for the explanation @AlonaKaplan , I indeed misunderstood the the origianal comment.
Sorry for misleading you @alonSadan!

Reply:
all good. I understood it the same way you did.